includes/class-wp-classroom-users.php

```diff
@@ -149,10 +149,14 @@ public function redirect()
     {
         global $post;
         $url = wp_login_url();
-
+        $all_terms = get_the_terms($post->ID, get_taxonomies())[0];
+        $course_id = $all_terms->parent ? $all_terms->parent : $all_terms->term_id;
```

Comment:
Why are we using get_taxonomies() ? This is only applicable to `wp_course`. Also, we're only checking the first term for some reason. There should probably be a loop somewhere to check presence of redirect on each term this class has.